--configure-on-demand results in obscure failures

## What happened?

With GCV 1.7.1, @uschi2000 found that running `./gradlew build --configure-on-demand` resulted in a strange looking error:

```
FAILURE: Build failed with an exception.

* What went wrong:
Cannot change dependencies of configuration ':issues:runtimeClasspath' after it has been resolved.

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.InvalidUserDataException: Cannot change dependencies of configuration ':issues:runtimeClasspath' after it has been resolved.
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.preventIllegalMutation(DefaultConfiguration.java:1067)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.validateMutation(DefaultConfiguration.java:1038)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.extendsFrom(DefaultConfiguration.java:361)
        at com.palantir.gradle.versions.VersionsLockPlugin.lambda$createTopConfiguration$41(VersionsLockPlugin.java:478)
        at org.gradle.api.internal.DefaultMutationGuard$2.execute(DefaultMutationGuard.java:41)
        at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction$1$1.run(DefaultCollectionCallbackActionDecorator.java:100)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.reapply(DefaultUserCodeApplicationContext.java:58)
        at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction$1.run(DefaultCollectionCallbackActionDecorator.java:97)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
```
## What did you want to happen?

Don't break, or at least tell me that I need to turn off `--configure-on-demand`.

--- a/src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java
+++ b/src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java
@@ -372,6 +372,11 @@ public class VersionsLockPlugin implements Plugin<Project> {
             throw new GradleException("Must be applied only to root project");
         }
 
+        Preconditions.checkState(
+                !project.getGradle().getStartParameter().isConfigureOnDemand(),
+                "Gradle Consistent Versions doesn't currently work with configure-on-demand, please remove"
+                        + " 'org.gradle.configureondemand' from your gradle.properties");
+
         project.subprojects(subproject -> {
             subproject.afterEvaluate(sub -> {
                 if (haveSameGroupAndName(project, sub)) {

